config/configauth/default_serverauthenticator.go

```diff
@@ -41,15 +40,15 @@ func WithAuthenticate(authenticateFunc AuthenticateFunc) Option {
 
 // WithStart overrides the default `Start` function for a component.Component.
 // The default always returns nil.
-func WithStart(startFunc componenthelper.StartFunc) Option {
+func WithStart(startFunc component.StartFunc) Option {
```

Comment:
Interesting, despite the package change, this isn't a breaking change, as what matters is the actual signature for the func. Looks good :-)